zerver/tests/test_auth_backends.py

```diff
@@ -2286,10 +2286,9 @@ def test_id_token_with_invalid_aud_sent(self) -> None:
             id_token=self.generate_id_token(account_data_dict, audience='com.different.app'),
         )
 
-        with self.apple_jwk_url_mock(), mock.patch('logging.info') as mock_info:
+        with self.apple_jwk_url_mock(),  self.assertLogs(level='INFO') as m:
             result = self.client_get(url, **headers)
-            mock_info.assert_called_once_with('/complete/apple/: %s',
-                                              'Authentication failed: Token validation failed')
+        self.assertEqual(m.output, ["INFO:root:/complete/apple/: Authentication failed: Token validation failed"])
```

Comment:
This is right for the current code. But I just saw that I missed using `self.logger` for the code this is testing.
Feel free change this https://github.com/zulip/zulip/blob/master/zproject/backends.py#L1696 line to use `self.logger` and the test code here accordingly.
(Keep this as a separate commit)

Reply:
made changes in new [commit](https://github.com/zulip/zulip/pull/15708/commits/6127d0f56bf41b360514fb634fbceafe159c8313)